File: JAVA/2583. Kth Largest Sum in a Binary Tree.java
class Solution {
    public long kthLargestLevelSum(TreeNode root, int k) {
      Queue<TreeNode> q = new LinkedList<>();
      q.add(root);
      List<Long> list = new ArrayList<>();
      while(!q.isEmpty())
      {
        int size = q.size();
        long total = 0;
        for(int i=0;i<size;i++)
        {
            TreeNode temp = q.poll(); 
            total+=temp.val;
            if(temp.left!=null)
            q.add(temp.left);
            if(temp.right!=null)
            q.add(temp.right);
        }
        list.add(total);
      }  
      if(k>list.size())
      return -1;
      Collections.sort(list);
      return list.get(list.size()-k);
    }
}

File: C++/2583. Kth Largest Sum in a Binary Tree.cpp
class Solution {
public:
    long long kthLargestLevelSum(TreeNode* root, int k) {
        long long sum=0;
        vector<long long> ans;
        queue<TreeNode*> q;
        if(!root)   
        return -1;
        q.push(root);
        while(!q.empty()){
            int n=q.size();
            sum=0;
            while(n--){
                TreeNode * temp=q.front();
                q.pop();
                sum+=temp->val;
                if(temp->left)
                q.push(temp->left);
                if(temp->right)
                q.push(temp->right);
            }
            ans.push_back(sum);
        }
        sort(ans.begin(),ans.end(),greater<long long >());
        if(ans.size()<k)
        return -1;
        else
        return ans[k-1];
    }
};
